perf/benchmark_setup_linux.sh

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env bash
+
+# Function to determine what is set in terms of NUMA and possibly others et al
+function checkHostReadiness() {
+  echo "=========================================================="
+  echo "Running numactl --show to determine if/how NUMA is enabled"
+  echo 
+  numactl --show
+  echo "=========================================================="
+}
+
+# Function to check and echo the current SMT state
+function checkSMTState() {
+    local smt_state
+    smt_state=$(cat /sys/devices/system/cpu/smt/control)
+    echo "Current SMT State: $smt_state"
+}
+
+# Function to temporarily disable SMT
+function disableSMT() {
+    echo "============================================================="
+    echo "Temporarily Disabling SMT (Simultaneous Multithreading)"
+    
+    # Echo the SMT state before disabling
+    echo "SMT state before change:"
+    checkSMTState
+
+    # Temporarily disable SMT (this does not persist after reboot)
+    echo off | sudo tee /sys/devices/system/cpu/smt/control > /dev/null
+    
+    # Echo the SMT state after disabling
+    echo "SMT state after temporary change:"
+    checkSMTState
+
+    echo "SMT has been temporarily disabled. This change will not persist after a reboot."
+    echo "============================================================="
+}
+
+# Function to install cpufrequtils and set CPU governor to performance
+function setCPUGovernorToPerformance() {
+    echo "============================================================="
+    echo "Installing cpufrequtils and setting CPU governor to performance"
+
+    # Install cpufrequtils
+    sudo apt-get install -y cpufrequtils
+    
+    # Echo the current CPU governor state before change
+    echo "Current CPU Governor State:"
+    cat /sys/devices/system/cpu/cpu*/cpufreq/scaling_governor
+
+    # Set the governor to "performance"
+    echo 'GOVERNOR="performance"' | sudo tee /etc/default/cpufrequtils > /dev/null
+
+    # Apply the performance governor setting
+    sudo systemctl restart cpufrequtils
```

Comment:
What does this reference to `restart` do?

Reply:
The sudo systemctl restart cpufrequtils command reloads the cpufrequtils service to ensure the new configuration is applied. This is necessary when changing the governor because the service manages CPU frequency scaling policies.